Src/Web/Web.Net45/AspNetDiagnosticTelemetryModule.cs

```diff
@@ -0,0 +1,211 @@
+﻿namespace Microsoft.ApplicationInsights.Web
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Web;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    /// <summary>
+    /// Listens to ASP.NET DiagnosticSource and enables instrumentation with Activity: let ASP.NET create root Activity for the request.
+    /// </summary>
+    public class AspNetDiagnosticTelemetryModule : IObserver<DiagnosticListener>, IDisposable, ITelemetryModule
+    {
+        private const string AspNetListenerName = "Microsoft.AspNet.TelemetryCorrelation";
```

Comment:
>Microsoft.AspNet.TelemetryCorrelation [](start = 51, length = 37)

will it report begin and end callback in future as well? Maybe RequestsTracking? RequestsTelemetryTracking? (this name is better than Diagnostics though)